rbi/annotations/activesupport.rbi

```diff
@@ -46,3 +46,71 @@ class Object
   sig { returns(T::Boolean) }
   def present?; end
 end
+
+class Hash
+  sig { returns(T::Boolean) }
+  def extractable_options?; end
+end
+
+class Array
+  sig { params(position: Integer).returns(T::Array[T.untyped]) }
+  def from(position); end
```

Comment:
We should be able to use either `returns(T.self_type)` or `returns(T::Array[Elem])` for this and most of the methods below.

For methods that return the same type as the receiver, `T.self_type` is more appropriate. For methods that return the type of the element, like `second` for example, using `Elem` is more appropriate.